model-optimizer/extensions/front/tf/CTCGreedyDecoderReplacement.py

```diff
@@ -14,67 +14,223 @@
+        unsqueeze_batch_size = create_op_with_const_inputs(graph, Unsqueeze, {1: int64_array(0)},
+                                                           {'name': strided_slice_name + '/Unsqueeze'})
+        dims.in_port(0).get_connection().set_destination(unsqueeze_batch_size.in_port(0))
+        unsqueeze_time_size = create_op_with_const_inputs(graph, Unsqueeze, {1: int64_array(0)},
+                                                           {'name': strided_slice_1_name + '/Unsqueeze'})
+        fill.in_port(1).get_connection().set_destination(unsqueeze_time_size.in_port(0))
+
+        # compute a sequence mask shape [T, N] required for CTCGreedyDecoder
+        seq_mask_shape = Concat(graph, {'axis': 0, 'in_ports_count': 2,
+                                        'name': ctc_greedy_decoder_name + '/SequenceMaskShape'}).create_node()
+        seq_mask_shape.in_port(0).connect(unsqueeze_time_size.out_port(0))
+        seq_mask_shape.in_port(1).connect(unsqueeze_batch_size.out_port(0))
+
+        # compute a sequence mask
+        sequence_mask = create_op_with_const_inputs(graph, Broadcast, {0: np.array([1.0], dtype=np.float)},
+                                                    {'mode': 'numpy',
+                                                     'name': ctc_greedy_decoder_name + '/SequenceMask'})
+        sequence_mask.in_port(1).connect(seq_mask_shape.out_port(0))
+
+        # create CTCGreedyDecoder with the sequence mask instead of sequence length
+        ctc_greedy_decoder.in_port(1).disconnect()
+        ctc_greedy_decoder.in_port(1).connect(sequence_mask.out_port(0))
+
+        # remove fill and pack nodes since they are now in unconnected component
+        graph.remove_nodes_from([fill.id, dims.id])
+
+        # transform opset CTCGreedyDecoder output to TensorFlow's one that has a shape [N, T]
+        # opset CTCGreedyDecoder has an output with a shape [N, T, 1, 1]
+        squeeze_dec_seq = create_op_with_const_inputs(graph, Squeeze, {1: int64_array([2, 3])},
+                                                      {'name': sparse_to_dense_name})
+        squeeze_dec_seq.in_port(0).connect(ctc_greedy_decoder.out_port(0))
+        cast_to_int = Cast(graph, {'name': sparse_to_dense_name + '/CastToInt',
+                                   'dst_type': np.int32}).create_node()
+        cast_to_int.in_port(0).connect(squeeze_dec_seq.out_port(0))
+
+        # preserve output name from original graph
+        rename_nodes([(sparse_to_dense, sparse_to_dense_name + '/AbandonedName'),
+                      (cast_to_int, sparse_to_dense_name)])
+
+        # set output of the new sub-graph as a source for SparseToDense consumer
+        sparse_to_dense.out_port(0).get_connection().set_source(cast_to_int.out_port(0))
+
+        # cleanup a graph: disconnect CTCGreedyDecoder from Cast and SparseToDense
+        # since opset CTCGreedyDecoder outputs decoded sequence in a dense format
+        cast.in_port(0).disconnect()
```

Comment:
I doubt that it is necessary to disconnect these ports since we remove these nodes 3 lines below. So, please, remove these lines.

Reply:
Fixed.